tools/codegen/api/translate.py

```diff
@@ -94,6 +94,12 @@ def translate(
             ctx[NamedCType(t.elem.elem.name, ConstRefCType(BaseCType(tensorT)))] = \
                 f'({b.expr}.has_value() ? *{b.expr} : at::Tensor())'
 
+        if isinstance(t, NamedCType) and isinstance(t.type, ConstRefCType) and \
+                isinstance(t.type.elem, OptionalCType) and isinstance(t.type.elem.elem, BaseCType) \
+                and str(t.type.elem.elem.type) == 'at::Tensor':
```

Comment:
I know you're copying the code above but you can actually do this much more straightforwardly: `t.type == ConstRefCType(OptionalCType(BaseCType(tensorT)))`